TELCODOCS-2045 - Telco core RDS 4.18 docs (#88622)
openshift/openshift-docs: modules/telco-core-nmstate-operator.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/telco_core_ref_design_specs/telco-core-rds.adoc

:_mod-docs-content-type: REFERENCE
[id="telco-core-nmstate-operator_{context}"]
= NMState Operator

New in this release::
* No reference design updates in this release

Description::

> The Kubernetes NMState Operator provides a Kubernetes API for performing state-driven network configuration across the cluster nodes with NMState.

Reviewer: Nodes with NMState??

It enables network interface configurations, static IPs and DNS, VLANs, trunks, bonding, static routes, MTU, and enabling promiscuous mode on the secondary interfaces.

> The cluster nodes periodically reports on the state of each node's network interfaces to the API server.

Reviewer 2: ```suggestion
The cluster nodes periodically report on the state of each node's network interfaces to the API server.
```

Limits and requirements::
Not applicable

Engineering considerations::
* Initial networking configuration is applied using `NMStateConfig` content in the installation CRs.
The NMState Operator is used only when required for network updates.

> * When SR-IOV virtual functions are used for host networking, the NMState Operator (via `nodeNetworkConfigurationPolicy` CRs) is used to configure those VF interfaces, for example, VLANs and MTU.

Reviewer 2: maybe to rephrase it to be clearer?
```suggestion
* When SR-IOV virtual functions are used for host networking, the NMState Operator (via `nodeNetworkConfigurationPolicy` CRs) is used to configure VF interfaces, such as VLANs and MTU.
```
